BUILDING.md

```diff
@@ -107,7 +107,8 @@ Therefore, gRPC supports several major build systems, which should satisfy most
 
 From the grpc repository root
 ```sh
- $ make
+ $ make 
+ $ make install
```

Comment:
this doesn't logically belong here. These are instructions how to build gRPC itself.

The instructions how to  use gRPC in your projects are here: https://github.com/grpc/grpc/tree/master/src/cpp#to-start-using-grpc-c (which involves the `make install` piece for make and also a warning what's bad about using it).

We can add a link to the "how to use" instructions from BUILDING.md, but we shouldn't add the `make install` command.

Reply:
Okay but it needs to be modified in BUILDING.md, it should be included as link. Cause build fails incase someone is trying to run grpc exxamples by just following instructions in the BUILDING.md file.

Reply:
Addressing in https://github.com/grpc/grpc/pull/16412.